cosmo/serializer.py

```diff
@@ -233,6 +243,28 @@ def serialize(self):
                     "route_distinguisher": "9136:" + str(l2vpn["identifier"]),
                     "vrf_target": "target:1:" + str(l2vpn["identifier"]),
                 }
+            elif l2vpn['type'] == "VPWS":
+                routing_instances[l2vpn["name"].replace("WAN: ", "")] = {
+                    "interfaces": [
+                        i["name"] for i in l2vpn["interfaces"]
+                    ],
+                    "description": "VPWS: " + l2vpn["name"].replace("WAN: VS_", ""),
+                    "instance_type": "evpn-vpws",
+                    "protocols": {
+                        "evpn": {
+                            "interfaces": {
+                                i["name"]: {
+                                    "vpws_service_id": {
+                                        "local": int(i['id']),
+                                        "remote": int([termination["assigned_object"]["id"] for termination in l2vpn["terminations"] if termination["assigned_object"]["id"] != i['id']][0]),
+                                    },
+                                } for i in l2vpn["interfaces"]
```

Comment:
You might want to to generatet this before the object declaration and use some auxiliary varialbes for better and more readable naming. Especially the `remote` key in `vpws_service_id`.. :laughing: